cannot access site

I update settings.json with the ep_oauth section (and the correct settings from github), but I am unable to access the site at all after it is active.

Here is my log after installing the plugin (with the ep_oauth present):

```
[2023-08-21 18:53:14.491] [INFO] plugins - Installing plugin ep_oauth...
[2023-08-21 18:53:25.704] [ERROR] runCmd|npm - npm WARN enoent ENOENT: no such file or directory, open '/opt/etherpad/package.json'
[2023-08-21 18:53:25.714] [ERROR] runCmd|npm - npm WARN etherpad No description
[2023-08-21 18:53:25.721] [ERROR] runCmd|npm - npm WARN etherpad No repository field.
[2023-08-21 18:53:25.726] [ERROR] runCmd|npm - npm WARN etherpad No README data
[2023-08-21 18:53:25.730] [ERROR] runCmd|npm - npm WARN etherpad No license field.
[2023-08-21 18:53:25.731] [ERROR] runCmd|npm -
[2023-08-21 18:53:25.793] [INFO] runCmd|npm - + ep_oauth@0.0.18
[2023-08-21 18:53:25.794] [INFO] runCmd|npm - added 1 package from 1 contributor and audited 1348 packages in 10.782s
[2023-08-21 18:53:26.688] [INFO] runCmd|npm -
[2023-08-21 18:53:26.689] [INFO] runCmd|npm - 209 packages are looking for funding
[2023-08-21 18:53:26.689] [INFO] runCmd|npm -   run `npm fund` for details
[2023-08-21 18:53:26.689] [INFO] runCmd|npm -
[2023-08-21 18:53:26.691] [INFO] runCmd|npm - found 28 vulnerabilities (20 moderate, 8 high)
[2023-08-21 18:53:26.691] [INFO] runCmd|npm -   run `npm audit fix` to fix them, or `npm audit` for details
[2023-08-21 18:53:26.727] [INFO] plugins - Successfully installed plugin ep_oauth
[2023-08-21 18:53:26.727] [INFO] plugins - Running npm to get a list of installed plugins...
[2023-08-21 18:53:32.899] [INFO] plugins - Loading plugin ep_adminpads2...
[2023-08-21 18:53:32.906] [INFO] plugins - Loading plugin ep_code_formatting...
[2023-08-21 18:53:32.906] [INFO] plugins - Loading plugin ep_etherpad-lite...
[2023-08-21 18:53:32.906] [INFO] plugins - Loading plugin ep_oauth...
[2023-08-21 18:53:32.907] [INFO] plugins - Loaded 4 plugins
[2023-08-21 18:53:33.042] [INFO] settings - settings loaded from: /opt/etherpad/settings.json
[2023-08-21 18:53:33.043] [INFO] settings - No credentials file found in /opt/etherpad/credentials.json. Ignoring.
[2023-08-21 18:53:33.044] [WARN] settings - The logconfig setting is deprecated.
[2023-08-21 18:53:33.045] [INFO] settings - Using skin "colibris" in dir: /opt/etherpad/src/static/skins/colibris
[2023-08-21 18:53:33.046] [INFO] settings - Random string used for versioning assets: 0134d049
[2023-08-21 18:53:33.046] [INFO] http - Closing HTTP server...
[2023-08-21 18:53:33.047] [INFO] socket.io - Closing socket.io engine...
[2023-08-21 18:53:33.047] [INFO] socket.io - All socket.io clients have disconnected
[2023-08-21 18:53:33.048] [INFO] http - Waiting for 1 HTTP clients to disconnect...
[2023-08-21 18:53:33.109] [INFO] http - HTTP server closed
```

after this, i try to access http://mysite:9001/admin or http://mysite:9001/p/aPadThatExists, but I just get "unable to connect".  

am I missing any steps?

--- a/auth.js
+++ b/auth.js
@@ -8,15 +8,29 @@ const request = require('request');
 /* eslint-disable-next-line node/no-unpublished-require */
 const OAuth2 = require('oauth').OAuth2;
 
-// Setup the oauth2 connector -- Doesn't establish any connections etc.
-const oauth2 = new OAuth2(settings.ep_oauth.clientID,
-    settings.ep_oauth.clientSecret,
-    'https://github.com/',
-    'login/oauth/authorize',
-    'login/oauth/access_token',
-    null); /** Custom headers */
+// If the plugin is installed but the operator hasn't added the
+// `ep_oauth` block to settings.json yet, reading `.clientID` on an
+// undefined object used to crash Etherpad at startup with a cryptic
+// "Cannot read properties of undefined" (that's what produced the "can't
+// access site" report in #63). Bail cleanly with a warning instead so
+// Etherpad keeps running; the plugin's hooks self-disable when the
+// client isn't configured.
+let oauth2 = null;
+if (settings.ep_oauth && settings.ep_oauth.clientID && settings.ep_oauth.clientSecret) {
+  // Setup the oauth2 connector -- Doesn't establish any connections etc.
+  oauth2 = new OAuth2(settings.ep_oauth.clientID,
+      settings.ep_oauth.clientSecret,
+      'https://github.com/',
+      'login/oauth/authorize',
+      'login/oauth/access_token',
+      null); /** Custom headers */
+} else {
+  console.warn(
+      'ep_oauth: missing settings.ep_oauth.clientID / clientSecret — plugin disabled');
+}
 
 exports.expressConfigure = (hookName, args, cb) => {
+  if (!oauth2) return cb && cb();
   // args.app.get('/auth/callback', function(req, res){
 
   // THIRD STEP
@@ -92,6 +106,7 @@ exports.expressConfigure = (hookName, args, cb) => {
 
 // FIRST STEP
 exports.authorize = (hookName, args, cb) => {
+  if (!oauth2) return cb([true]); // plugin disabled, don't block the request
   // Never lands here for url /auth/callback
   if (args.req.url.indexOf('/auth') === 0) return cb([true]);
 
@@ -104,6 +119,7 @@ exports.authorize = (hookName, args, cb) => {
 
 // SECOND STEP
 exports.authenticate = (hookName, args, cb) => {
+  if (!oauth2) return cb([]); // plugin disabled, defer to other auth plugins
   console.debug(`Database Write -> oauthredirectlookup:${args.req.sessionID}`, '---', args.req.url);
   db.set(`oauthredirectlookup:${args.req.sessionID}`, args.req.url);
   // User is not authorized so we need to do the authentication step

